server/src/visitors/YmlFunctionVisitor.ts

```diff
@@ -1,21 +1,62 @@
 import { CompletionItemKind } from "vscode-languageserver";
 import { YmlCompletionItemsProvider } from "../completion/YmlCompletionItemsProvider";
+import { YmlDefinitionProvider } from "../definitions";
 import { FunctionContext } from "../grammar";
-import { createNewCompletionItem } from "./VisitorsUtils";
+import { createLocation, createNewCompletionItem } from "./VisitorsUtils";
 import { YmlBaseVisitor } from "./YmlBaseVisitor";
 
 export class YmlFunctionVisitor extends YmlBaseVisitor {
-  constructor(completionProvider: YmlCompletionItemsProvider, uri: string) {
+  constructor(
+    completionProvider: YmlCompletionItemsProvider,
+    uri: string,
+    public definitions: YmlDefinitionProvider,
+  ) {
     super(completionProvider, uri);
   }
   public visitFunction(node: FunctionContext): void {
-    createNewCompletionItem(
-      this.uri,
-      this.completionProvider,
-      node.ymlId(),
-      node.field(),
-      CompletionItemKind.Function,
-      null,
+    const functionName = node.ymlId().text;
+    if (!this.isMethodInstanciation(functionName)) {
+      createNewCompletionItem(
+        this.uri,
+        this.completionProvider,
+        node.ymlId().text,
+        node.field(),
+        CompletionItemKind.Function,
+        null,
+      );
+      this.definitions.addDefinition(
+        createLocation(node.ymlId().text, node.start, node.stop, this.uri),
+      );
+    } else {
+      /*
+       * The function is a method instance.
+       * It has already been added as a completion item and a definition by YmlClassVisitor.
+       */
+    }
+  }
+
+  /**
+   * Check if the provided function name is the name of a class method instanciation.
+   * If the function name is the name of a class method instanciation:
+   * - it follows the format “className” + “::” + “methodName”
```

Comment:
* Grr, hyphens?
* Don't use homemade lists in doc since the newlines are prone to be ignored by IDEs. Use `<ul>` and stuff. But I'm not even sure those are supported in JS, so consider rephrasing stuff to get rid of the list altogether.

… Actually, that _would_ work if this was parsed like some kind of Markdown or whatever. What does your IDE display for this version of the comment?

Reply:
This *is* Markdown.
![capture d ecran de 2019-02-28 15-15-36](https://user-images.githubusercontent.com/15327919/53572553-ce49e700-3b6b-11e9-88bb-911e19cabaf0.png)

Reply:
Hum.